java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -304,19 +306,25 @@ private boolean shouldTerminate(Set<LearnedConstraint> learnedConstraints) {
 
     List<JavaFileScannerContext.Location> flowFromLearnedAssociation(LearnedAssociation learnedAssociation, ExplodedGraph.Node node) {
       ImmutableList.Builder<JavaFileScannerContext.Location> flowBuilder = ImmutableList.builder();
-      PMap<Class<? extends Constraint>, Constraint> allConstraints = node.programState.getConstraints(learnedAssociation.symbolicValue());
-      Collection<Constraint> constraints = filterByDomains(allConstraints);
-      for (Constraint constraint: constraints) {
-        String symbolName = learnedAssociation.symbol().name();
-        String msg;
-        if (ObjectConstraint.NULL == constraint && assigningNullFromTernary(node)) {
-          msg = String.format(IMPLIES_CAN_BE_NULL_MSG, symbolName);
-        } else if (assigningFromMethodInvocation(node) && assignedFromYieldWithUncertainResult(constraint, node)) {
-          msg = String.format("Implies '%s' can be %s.", symbolName, constraint.valueAsString());
-        } else {
-          msg = String.format("'%s' is assigned %s.", symbolName, constraint.valueAsString());
+      ProgramState programState = node.programState;
+      Symbol rhsSymbol = symbolFromStack(learnedAssociation.symbolicValue(), programState);
+      if (rhsSymbol != null) {
+        flowBuilder.add(location(node, String.format(IMPLIES_SAME_VALUE, learnedAssociation.symbol().name(), rhsSymbol.name())));
+      } else {
+        PMap<Class<? extends Constraint>, Constraint> allConstraints = programState.getConstraints(learnedAssociation.symbolicValue());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "programState" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)